backend/inspirehep/search/facets.py

```diff
@@ -310,16 +339,36 @@ def citations_by_year():
         for key, val in current_app.config["HEP_COMMON_FILTERS"].items()
         if key not in excluded_filters
     }
+    map_script = """
+        def add(def x, def y) {
+            x + y
+        }
+        def years = params._source.citations_by_year != null ? params._source.citations_by_year : [];
+        for (element in years) {
+            state.merge(element.year.toString(), element.count, this::add)
```

Comment:
I think it would have looked nicer to use a lambda here instead of ```this:add``` since you are defining add in both places.

Reply:
I agree. I initially tried 
```
            state.merge(element.year.toString(), element.count, (old, new) -> old + new)
```
which looks a lot like the 4th example in https://www.elastic.co/guide/en/elasticsearch/painless/7.1/painless-lambdas.html but couldn't make it work (syntax error on the `new` in the lambda argument) so used this as an alternative.

Reply:
I think it is because ```new``` is a reserved word, should have worked otherwise

Reply:
:man_facepalming: